### PR TITLE
chore: Address linter issues in e2e/

### PR DESCRIPTION
**Pull request by bznein**
## Description

Addresses linting issues (mostly unchecked type assertions) in e2e.

Upgrading to a newer version of the linter must be done at the very end since it raises hundreds of errors. 
I am going to incrementally address them and only upgrade the version when they are all solved.

ref: #6086 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [x] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Refactor**
	- Improved type safety in various test suites by adding type assertion checks.
	- Updated import paths for enhanced project structure.
- **Bug Fixes**
	- Fixed potential issues in handling type conversions and assertions to prevent runtime errors.

### CODE REVIEW

File: e2e/tests/transfer/localhost_test.go [hunk 1]
```diff
@@ -56,7 +56,8 @@ func (s *LocalhostTransferTestSuite) TestMsgTransfer_Localhost() {
 		cs, err := s.QueryClientState(ctx, chainA, exported.LocalhostClientID)
 		s.Require().NoError(err)
 
-		localhostClientState := cs.(*localhost.ClientState)
+		localhostClientState, ok := cs.(*localhost.ClientState)
+		s.Require().True(ok)
```
[the remark this answers is not in the document]
Comment (bznein) [reply e2e/tests/transfer/localhost_test.go:60]: We use s.Require() with no message extensively. I'm open to change if any reviewer agrees with the rabbit :)

File: e2e/testsuite/testconfig.go [hunk 2]
```diff
@@ -741,7 +741,10 @@ func modifyChannelGenesisAppState(ibcAppState []byte) ([]byte, error) {
 	}
 
 	// be ashamed, be very ashamed
-	channelGenesis := ibcGenesisMap["channel_genesis"].(map[string]interface{})
+	channelGenesis, ok := ibcGenesisMap["channel_genesis"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("can't convert ibcGenesisMap entry into map[string]interface{} type")
```
Comment (crodriguezvega) [inline e2e/testsuite/testconfig.go:746]: For this kind of errors we would usually do something like...

```suggestion
		return nil, fmt.Errorf("can't convert IBC genesis map entry into type %T", &map[string]interface{})
```

**Review by crodriguezvega — APPROVED** [review]
Thanks for working on all these listing fixes, @bznein. :)

File: e2e/testsuite/testconfig.go (hunk 2 above)
Comment (bznein) [reply e2e/testsuite/testconfig.go]: Done, though I had to do a small workaround as it won't work by putting directly `map[string]interface{}` there

Error:
```
codec.go:1: : # github.com/cosmos/ibc-go/e2e/testsuite
./testconfig.go:746:79: map[string]interface{} (type) is not an expression (typecheck)
```

**Review by chatton — APPROVED** [review]

**Pull request merged**